File: backend/app/schemas/medical_record.py
from datetime import date
from typing import Optional

from pydantic import BaseModel


class MedicalRecordBase(BaseModel):
    title: str
    record_type: str

    hospital_name: Optional[str] = None
    doctor_name: Optional[str] = None

    visit_date: Optional[date] = None

    notes: Optional[str] = None

    file_name: Optional[str] = None


class MedicalRecordCreate(MedicalRecordBase):
    pass


class MedicalRecordUpdate(BaseModel):
    title: Optional[str] = None
    record_type: Optional[str] = None

    hospital_name: Optional[str] = None
    doctor_name: Optional[str] = None

    visit_date: Optional[date] = None

    notes: Optional[str] = None

    file_name: Optional[str] = None


class MedicalRecordResponse(MedicalRecordBase):
    id: int
    user_id: int

    model_config = {
        "from_attributes": True
    }

File: frontend/src/types/medicalRecord.ts
export interface MedicalRecord {
  id: number
  user_id: number
  title: string
  record_type: string
  hospital_name?: string | null
  doctor_name?: string | null
  visit_date?: string | null
  notes?: string | null
  file_name?: string | null
}

export interface MedicalRecordCreate {
  title: string
  record_type: string
  hospital_name?: string | null
  doctor_name?: string | null
  visit_date?: string | null
  notes?: string | null
  file_name?: string | null
}

export type MedicalRecordUpdate = Partial<MedicalRecordCreate>
